engine/reformulation/core/src/main/java/it/unibz/inf/ontop/answering/resultset/GraphResultSet.java

```diff
@@ -1,10 +1,16 @@
 package it.unibz.inf.ontop.answering.resultset;
 
+import it.unibz.inf.ontop.answering.resultset.impl.RDFFactCloseableIterator;
+import it.unibz.inf.ontop.exception.OntopConnectionException;
+
 import it.unibz.inf.ontop.exception.OntopQueryAnsweringException;
 import it.unibz.inf.ontop.spec.ontology.RDFFact;
 
 public interface GraphResultSet<X extends OntopQueryAnsweringException> extends IterativeOBDAResultSet<RDFFact, X> {
 
     @Override
-    RDFFact next() throws X;
+    RDFFact next() throws X, OntopConnectionException;
+
+    RDFFactCloseableIterator iterator();
```

Comment:
`RDFFactCloseableIterator` is not an interface. Please use `OntopCloseableIterator<RDFFact, OntopConnectionException>`